Source: Internal

Info: This patch makes the changes necessary to support wanboot.
Mostly it removes floating point support.

Status: This patch is Solaris specific.  This will not be sent upstream.

--- gnutls-3.4.6/gl/printf-args.c 2015-01-19 09:38:09.000000000 -0800
+++ gnutls-3.4.6/gl/printf-args.c 2015-12-10 17:46:30.949444885 -0800
@@ -73,12 +73,14 @@
         ap->a.a_ulonglongint = va_arg (args, unsigned long long int);
         break;
 #endif
+#ifndef _BOOT
       case TYPE_DOUBLE:
         ap->a.a_double = va_arg (args, double);
         break;
       case TYPE_LONGDOUBLE:
         ap->a.a_longdouble = va_arg (args, long double);
         break;
+#endif /* !_BOOT */
       case TYPE_CHAR:
         ap->a.a_char = va_arg (args, int);
         break;
--- gnutls-3.4.6/src/gl/timespec.h 2015-01-19 09:38:09.000000000 -0800
+++ gnutls-3.4.6/src/gl/timespec.h 2015-12-10 17:47:59.802282095 -0800
@@ -97,12 +97,14 @@
 struct timespec dtotimespec (double)
   _GL_ATTRIBUTE_CONST;
 
+#ifndef _BOOT
 /* Return an approximation to A, of type 'double'.  */
 _GL_TIMESPEC_INLINE double
 timespectod (struct timespec a)
 {
   return a.tv_sec + a.tv_nsec / 1e9;
 }
+#endif /* !_BOOT */
 
 void gettime (struct timespec *);
 int settime (struct timespec const *);
--- gnutls-3.4.6/src/gl/printf-args.c 2015-01-19 09:38:09.000000000 -0800
+++ gnutls-3.4.6/src/gl/printf-args.c 2015-12-10 17:51:12.619526033 -0800
@@ -73,12 +73,14 @@
         ap->a.a_ulonglongint = va_arg (args, unsigned long long int);
         break;
 #endif
+#ifndef _BOOT
       case TYPE_DOUBLE:
         ap->a.a_double = va_arg (args, double);
         break;
       case TYPE_LONGDOUBLE:
         ap->a.a_longdouble = va_arg (args, long double);
         break;
+#endif /* !_BOOT */
       case TYPE_CHAR:
         ap->a.a_char = va_arg (args, int);
         break;
